src/typechecker.erl

```diff
@@ -1288,14 +1294,20 @@ subst_ty(_, Ty) -> Ty.
 %% Returns the type of the expression, a collection of variables bound in
 %% the expression together with their type and constraints.
 %-spec type_check_expr(#env, any()) -> { any(), #{ any() => any()}, #{ any() => any()} }.
-type_check_expr(Env, {var, P, Var}) ->
+type_check_expr(Env, Expr) ->
+    Res = {Ty, _VarBinds, _Cs} = do_type_check_expr(Env, Expr),
+    ?verbose(Env, "~s: Inferred type of ~s :: ~s~n",
```

Comment:
We only "infer" when the `infer` option is on. Otherwise, I think we should use the word "propagate" here.